snap-build/lib.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+error(){
+	msg="$*"
+	echo "ERROR: $msg" >&2
+}
+
+die(){
+	error "$*"
+	exit 1
+}
+
+make_random_ip_addr() {
+	echo "127.$((1 + RANDOM % 240)).$((1 + RANDOM % 240)).$((1 + RANDOM % 240))"
+}
+
+make_random_port() {
+	echo "$((11060 + RANDOM % 1000))"
+}
+
+get_dnssearch() {
+	echo "$(grep search /etc/resolv.conf | cut -d' ' -f 2)"
+}
+
+get_dns() {
+	v="$(grep nameserver /etc/resolv.conf | cut -d' ' -f2 | sed -e 's/^/"/g' -e 's/$/",/g')"
+	echo ${v} | sed -e 's|,$||g'
+}
+
+download() {
+	url="$1"
+	outdir="$2"
+	pushd "${outdir}"
+	curl -LO ${url}
+	status=$?
+	popd
+	return ${status}
+}
+
+setup_image() {
+	img_url=$1
+	img=$2
+	[ -f "${img}" ] && return
+	download "${img_url}" "$(dirname ${img})"
+	[ $? != 0 ] && rm -f "${img}" && return
```

Comment:
Is this going to work? The files which source `.ci/lib.sh` call `set -e` so if `curl` fails, the script will immediately exit.

Reply:
good catch!

Reply:
removed `set -e` since I have to check exit codes and poweroff the VM

Reply:
Note that you can do this:

```
set -e

{ download "${img_url}" "$(dirname ${img})"; ret=$?; } || true
[ $ret != 0 ] && rm -f "${img}" && return
```

Reply:
got it, thanks